java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1997,71 +1998,87 @@ public final Builder append(double value) {
       return this;
     }
 
-    public final Builder append(String value) {
+    public Builder append(String value) {
       assert value != null : "appendNull must be used to append null strings";
       return appendUTF8String(value.getBytes(StandardCharsets.UTF_8));
     }
 
-    private Builder appendUTF8String(byte[] value) {
+    public Builder appendUTF8String(byte[] value) {
       return appendUTF8String(value, 0, value.length);
     }
 
-    private Builder appendUTF8String(byte[] value, int offset, int length) {
+    public Builder appendUTF8String(byte[] value, int offset, int length) {
       assert value != null : "appendNull must be used to append null strings";
       assert offset >= 0;
       assert length >= 0;
       assert value.length + offset <= length;
       assert type == DType.STRING_CATEGORY || type == DType.STRING;
       assert currentIndex < rows;
-      assert (currentStringByteIndex + length) <= stringBufferSize;
+      // just for strings we want to throw a real exception if we would overrun the buffer
+      while (currentStringByteIndex + length > data.getLength()) {
+        long oldlen = data.getLength();
+        long newlen = oldlen * 2;
+        // need to grow the size of the buffer.
+        HostMemoryBuffer newData = HostMemoryBuffer.allocate(newlen);
+        try {
+          newData.copyFromHostBuffer(0, data, 0, currentStringByteIndex);
```

Comment:
Why allocate and copy all the data when we might have to do this again if the `while` loop doesn't terminate?  Ideally this should only allocate and copy once.

Reply:
You are right, and there is a corner case here that I have to address.